src/main/resources/db/migration/V1__create_tables.sql

```diff
@@ -57,7 +57,7 @@ CREATE TABLE job_run_states (
 
 CREATE TABLE datasets (
   id              SERIAL PRIMARY KEY,
-  name            VARCHAR(64) NOT NULL,
+  name            VARCHAR(64) UNIQUE NOT NULL,
```

Comment:
is `name` meant to be fairly free-form, or will it get standardized to something like NAMESPACE.NAME

Reply:
another question for thought, should the uniqueness of a dataset include it's db as well? <db_id, name>. Or if namespace is more explicit: <db_id, db_namespace, name>. Just ideas :)

Reply:
> is name meant to be fairly free-form ...

I think simple rules around name validation would make sense (ie allowed characters). We'll want dataset names to be human-readable / fairly free-form, but leaving general naming conventions up to the owner of the dataset.

> or will it get standardized to something like NAMESPACE.NAME ...

We'll want to standardize on some internal representations of datasets. Let's say we have an owner `analytics`. There should be only a single owner registered with marquez under that name. But datasets names shouldn't be globally unique, though unique to the owner.

**Valid**
*  `analytics.room_bookings`
*  `datascience.room_bookings`

Reply:
Cool, sounds good. 👍

Reply:
> another question for thought, should the uniqueness of a dataset include it's db as well? <db_id, name>. Or if namespace is more explicit: <db_id, db_namespace, name>

Agreed. Defining a tuple that uniquely identifies DBs totally makes sense (or at least some variation of what you are proposing).